astropy/modeling/mappings.py

```diff
@@ -55,17 +73,37 @@ def inverse(self):
                 "Mappings such as {0} that drop one or more of their inputs "
                 "are not invertible at this time.".format(self.mapping))
 
-        return self.__class__(mapping)
+        inv = self.__class__(mapping)
+        inv._inputs = self._outputs
+        inv._outputs = self._inputs
```

Comment:
I think this workaround looks reasonable now.  I agree there's some ambiguity in the definition (recall I originally wanted something stricter, but it was hard to come up with a good way, so this was good enough for now).